src/variableMenu.ts

```diff
@@ -0,0 +1,98 @@
+}
+
+function turnOnFormat(key: string) {
+    const debugSettingsRoot: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("java.debug.settings");
+    if (vscode.debug.activeDebugSession && vscode.debug.activeDebugSession.type === "java") {
+        const formatter: any = {
+            showHex: debugSettingsRoot.showHex,
+            showQualifiedNames: debugSettingsRoot.showQualifiedNames,
+            showStaticVariables: debugSettingsRoot.showStaticVariables,
+            showLogicalStructure: debugSettingsRoot.showLogicalStructure,
+            showToString: debugSettingsRoot.showToString,
+        };
+        formatter[key] = true;
+        vscode.debug.activeDebugSession.customRequest("refreshVariables", formatter);
+    }
+
+    // Update the formatter to settings.json
+    const inspect = vscode.workspace.getConfiguration("java.debug").inspect("settings");
+    let configurationTarget = vscode.ConfigurationTarget.Global;
+    if (inspect && inspect.workspaceFolderValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.WorkspaceFolder;
+    } else if (inspect && inspect.workspaceValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.Workspace;
+    }
+    debugSettingsRoot.update(key, true, configurationTarget);
+}
+
+function turnOffFormat(key: string) {
+    const debugSettingsRoot: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("java.debug.settings");
+    if (vscode.debug.activeDebugSession && vscode.debug.activeDebugSession.type === "java") {
+        const formatter: any = {
+            showHex: debugSettingsRoot.showHex,
+            showQualifiedNames: debugSettingsRoot.showQualifiedNames,
+            showStaticVariables: debugSettingsRoot.showStaticVariables,
+            showLogicalStructure: debugSettingsRoot.showLogicalStructure,
+            showToString: debugSettingsRoot.showToString,
+        };
+        formatter[key] = false;
+        vscode.debug.activeDebugSession.customRequest("refreshVariables", formatter);
+    }
+
+    // Update the formatter to settings.json
+    const inspect = vscode.workspace.getConfiguration("java.debug").inspect("settings");
+    let configurationTarget = vscode.ConfigurationTarget.Global;
+    if (inspect && inspect.workspaceFolderValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.WorkspaceFolder;
+    } else if (inspect && inspect.workspaceValue !== undefined) {
+        configurationTarget = vscode.ConfigurationTarget.Workspace;
+    }
+    debugSettingsRoot.update(key, false, configurationTarget);
+}
+
+function updateContextKeys() {
+    const debugSettingsRoot: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration("java.debug.settings");
+    if (debugSettingsRoot) {
+        vscode.commands.executeCommand("setContext", "java.debug.showHex", debugSettingsRoot.showHex ? "on" : "off");
```

Comment:
Will you consider to align with the context key format used in vscode java? https://github.com/redhat-developer/vscode-java/blob/master/package.json#L996

`[extension]:[key]`

Leave the choice to you since it's a personal preference

Reply:
nice, suggestions accepted.